src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -39,9 +39,14 @@ public String getToken() {
     // Called from other synchronized code. Do not call into other synchronized code or you'll
     // risk deadlock
     public synchronized void setDistinctId(String distinctId) {
-        mUnseenSurveys.clear();
-        mUnseenNotifications.clear();
-        mDistinctId = distinctId;
+        if (mDistinctId == distinctId) {
```

Comment:
This could be also called from the client by a Mixpanel instance: `mMixpanelInstance.getPeople().setDistinctId(null)`. However, I think we should also let the user nullify the `distinctId` since it's the initial state for this variable and we do sanity checks when we use it (e.g. https://github.com/mixpanel/mixpanel-android/blob/master/src/main/java/com/mixpanel/android/mpmetrics/DecideChecker.java#L200).